x/checkpointing/proposal.go

```diff
@@ -192,8 +192,14 @@ func (h *ProposalHandler) findLastBlockHash(extendedVotes []abci.ExtendedVoteInf
 	blockHashes := make(map[string]int64, 0)
 	// Iterate over vote extensions and if they have a valid structure
 	// increase the voting power of the block hash they commit to
+	var totalPower int64 = 0
 	for _, vote := range extendedVotes {
+		// accumulate voting power from all the votes
+		totalPower += vote.Validator.Power
 		var ve ckpttypes.VoteExtension
+		if len(vote.VoteExtension) == 0 {
+			continue
+		}
 		if err := ve.Unmarshal(vote.VoteExtension); err != nil {
 			continue
 		}
```

Comment:
```suggestion
		}
		bHash, err := ve.BlockHash.Marshal()
		if err != nil {
		    continue
		}
		blockHashes[hex.EncodeToString(bHash)] += vote.Validator.Power
```

Just to avoid an extra panic. @RafilxTenfen is an alternative as well.